### PR TITLE
add jhipster.api-docs.management-include-pattern property

### PR DESCRIPTION
**Pull request by PierreBesson**
this property allows custom setting of the management include pattern to support advanced use cases such as using a custom server path.

### CODE REVIEW

**Review by DanielFran — APPROVED** [review]

File: jhipster-framework/src/main/java/tech/jhipster/config/apidoc/SpringfoxAutoConfiguration.java [hunk 1]
```diff
@@ -135,16 +135,13 @@ public JHipsterSpringfoxCustomizer jHipsterSpringfoxCustomizer() {
      * Springfox configuration for the management endpoints (actuator) OpenAPI docs.
      *
      * @param appName               the application name
-     * @param managementContextPath the path to access management endpoints
      * @return the Springfox configuration
      */
     @Bean
     @ConditionalOnClass(name = "org.springframework.boot.actuate.autoconfigure.web.server.ManagementServerProperties")
-    @ConditionalOnProperty("management.endpoints.web.base-path")
     @ConditionalOnExpression("'${management.endpoints.web.base-path}'.length() > 0")
```
Comment (DanielFran) [inline jhipster-framework/src/main/java/tech/jhipster/config/apidoc/SpringfoxAutoConfiguration.java:142]: Should be removed also, no?

Comment (clocken) [reply jhipster-framework/src/main/java/tech/jhipster/config/apidoc/SpringfoxAutoConfiguration.java:142]: @DanielFran and @PierreBesson On the contrary. Even the
```java
@ConditionalOnProperty("management.endpoints.web.base-path")
```
should stay in my opinion as this property hasn't been touched.
Fixed that here 0bc2cb9. Feel free to cherry pick.

Comment (clocken) [reply jhipster-framework/src/main/java/tech/jhipster/config/apidoc/SpringfoxAutoConfiguration.java:142]: @DanielFran Correct. But I think it does not make sense to initialize the Docket for the management apis if `management.endpoints.web.base-path` has not been set in the first place. That's what the `@ConditionalOnProperty("management.endpoints.web.base-path")` was originally intended for, I think. Should have made that clearer in my first comment. Am I missing something?

Comment (DanielFran) [reply jhipster-framework/src/main/java/tech/jhipster/config/apidoc/SpringfoxAutoConfiguration.java:142]: @clocken The property have by default `"/management/.*"`.
User can use the property to rename the path and include or not the context path (example: `"/admin/.*"` or `"/mycontext/admin/.*"`).
So it not depends anymore from the property `management.endpoints.web.base-path`

Comment (clocken) [reply jhipster-framework/src/main/java/tech/jhipster/config/apidoc/SpringfoxAutoConfiguration.java:142]: @DanielFran Ok, now I see. Was missing that the `managementContextPath` was set to that property value before, sorry. Then I agree with you that the second line `@ConditionalOnExpression("'${management.endpoints.web.base-path}'.length() > 0")` should be removed also.

Comment (clocken) [reply jhipster-framework/src/main/java/tech/jhipster/config/apidoc/SpringfoxAutoConfiguration.java:142]: @PierreBesson Done in a59a29c

Comment (DanielFran) [reply jhipster-framework/src/main/java/tech/jhipster/config/apidoc/SpringfoxAutoConfiguration.java:142]: @PierreBesson Can you confirm if it should be removed or not?

Comment (PierreBesson) [reply jhipster-framework/src/main/java/tech/jhipster/config/apidoc/SpringfoxAutoConfiguration.java:142]: Yes it should be removed ! Sorry for the super late reply.

**Comment by pascalgrimaud** [comment]
@PierreBesson @DanielFran : should it be merged for v7 beta ?

**Comment by DanielFran** [comment]
@pascalgrimaud Not sure, I believe @PierreBesson did not finish yet to test the changes.

**Comment by PierreBesson** [comment]
> @PierreBesson @DanielFran : should it be merged for v7 beta ?

I would really like if it could but not mandatory.

**Comment by clocken** [comment]
> I would really like if it could but not mandatory.

@PierreBesson @pascalgrimaud Yeah, me too but unfortunately it seems we're not done just yet. I'm currently testing the cases mentioned [here](https://github.com/jhipster/generator-jhipster/pull/12923#issuecomment-744066353) and as @ruddell [pointed out](https://github.com/jhipster/generator-jhipster/pull/12923#issuecomment-727036042) page refreshing does not work currently with paths containing multiple slashes, thanks for that by the way. So far the only solution I can see is having to set the `base href` to the context path like before. I keep testing and maybe we can come up with another solution :thinking: If not I'll provide the necessary reverts and update the PR for the docs. At least the SwaggerUI will be fixed by the current changes.

**Comment by clocken** [comment]
@PierreBesson See [here](https://github.com/jhipster/generator-jhipster/pull/12923#issuecomment-748077732) for the `base href` fix. You also may cherry-pick [a59a29c](https://github.com/clocken/jhipster-bom/commit/a59a29ca1fe93705adf85327d70214b0728e0881) to remove that `@ConditionalOnExpression` from the review above.

**Review by clocken — APPROVED** [review]

**Comment by mraible** [comment]
Yay or nay @PierreBesson?

**Comment by PierreBesson** [comment]
I really want to have this for v7. I'm going to rebase, check that it still work and push for the merge.

**Pull request merged**